### PR TITLE
SelectObjectContent API with args & response object.

### PR DESCRIPTION
**Pull request by BigUstad**
Functional Test, Example changed accordingly.

### CODE REVIEW

File: Minio/DataModel/ObjectArgs.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        }
+
+        public override void Validate()
+        {
+            base.Validate();
+            utils.ValidateObjectName(this.ObjectName);
+        }
+
+        public Dictionary<string, string> MergedHeaders()
+        {
```
Comment (poornas) [inline Minio/DataModel/ObjectArgs.cs:93]: if ((this.ExtraHeaders == null )  
            {
                return this.HeaderMap;
            }
           if  (this.HeaderMap == null)
           {
               return this.ExtraHeaders;
           }
          // merge headers

File: Minio/MinioClient.cs [hunk 2]
```diff
@@ -169,6 +169,27 @@ private void ArgsCheck(Args args)
             return args.BuildRequest(request);
         }
 
+
+        /// <summary>
+        /// Constructs a RestRequest using bucket/object names from Args.
+        /// Calls overloaded CreateRequest method.
+        /// </summary>
+        /// <param name="args">The child object of BucketArgs class, args with populated values from Input</param>
```
Comment (poornas) [inline Minio/MinioClient.cs:177]: description of args needs correction

**Review by poornas — CHANGES_REQUESTED** [review]

File: Minio/DataModel/ObjectArgs.cs (hunk 1 above)
Comment (BigUstad) [reply Minio/DataModel/ObjectArgs.cs:94]: Change made.

File: Minio/MinioClient.cs (hunk 2 above)
Comment (BigUstad) [reply Minio/MinioClient.cs]: Changed the description.

File: Minio/DataModel/ObjectArgs.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        public override void Validate()
+        {
+            base.Validate();
+            utils.ValidateObjectName(this.ObjectName);
+        }
+
+        // Merge the Headers map & extra headers.
+        public Dictionary<string, string> MergedHeaders()
+        {
+            if ((this.ExtraHeaders == null && this.HeaderMap == null) ||
```
Comment (poornas) [inline Minio/DataModel/ObjectArgs.cs:95]: this check is not required - it is already taken care of if both are null by line 102

Comment (BigUstad) [reply Minio/DataModel/ObjectArgs.cs:95]: Yes. Removed it.

**Review by poornas — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]

**Pull request merged**